packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.test.js

```diff
@@ -111,15 +111,100 @@ describe('<TrapFocus />', () => {
       keyCode: 9, // Tab
     });
 
-    expect(document.querySelector('[data-test="sentinelStart"]')).toHaveFocus();
+    expect(screen.getByText('x')).toHaveFocus();
 
     initialFocus.focus();
     fireEvent.keyDown(screen.getByTestId('modal'), {
       keyCode: 9, // Tab
       shiftKey: true,
     });
 
-    expect(document.querySelector('[data-test="sentinelEnd"]')).toHaveFocus();
+    expect(screen.getByText('ok')).toHaveFocus();
+  });
+
+  it('should focus on first focus element after last has received a tab click', () => {
+    render(
+      <TrapFocus {...defaultProps} open>
+        <div tabIndex={-1} data-testid="modal">
+          <div>Title</div>
+          <button type="button">x</button>
+          <button type="button">cancel</button>
+          <button type="button">ok</button>
+        </div>
+      </TrapFocus>,
+    );
+    fireEvent.keyDown(screen.getByTestId('modal'), {
+      keyCode: 9, // Tab
+    });
+    userEvent.tab();
+    userEvent.tab();
+    userEvent.tab();
+    expect(screen.getByText('x')).toHaveFocus();
```

Comment:
It's not obvious that we need 3 tabs. The 2 last tabs could be no-ops and the test would still pass. Maybe all 3 are. Having an assertion after each tab would improve readability. 

Especially for focus behavior I'd rather have more assertions then absolutely necessary because you'll need them when working on the component later anyway.

Reply:
True, I'll assert after each tab so we have visibility into each event.

Reply:
> It's not obvious that we need 3 tabs.

The tab() calls have no effect. I have tested without.

It seems that we need to provide the trap focus root: https://github.com/testing-library/user-event/blob/a3f751f0fdfa514ac4e6723a6dfd13c5c2f68c6d/src/tab.js#L22.